packages/visual-editor/src/components/Directory.tsx

```diff
@@ -175,13 +180,23 @@ const DirectoryCard = ({
   cardNumber,
   profile,
   cardStyles,
+  puck,
 }: {
   cardNumber: number;
   profile: any;
   cardStyles: DirectoryProps["styles"]["cards"];
+  puck: PuckContext;
 }) => {
-  const { relativePrefixToRoot } = useTemplateProps();
+  const { document: streamDocument, relativePrefixToRoot } = useTemplateProps();
   const { i18n } = useTranslation();
+  const locale = i18n.language;
+
+  const resolvedUrl = resolveUrlTemplate(
+    streamDocument,
+    locale,
+    relativePrefixToRoot ?? "",
```

Comment:
Can you just pass relativePrefixToRoot and have it handled in resolveUrlTemplate?